packages/integration-test/src/setup.js

```diff
@@ -0,0 +1,242 @@
+     * such as sender, publicOwner and delegatedAddress
+     *
+     * @returns {Object} sender, publicOwner, delegatedAddress, opts variables
+     */
+    getTransactionTestingAddresses() {
+        const { sender, publicOwner, delegatedAddress, opts } = this;
+
+        return { sender, publicOwner, delegatedAddress, opts };
+    }
+
+    /**
+     * @method initialiseConfig - define and set the configuration to be used for the integration test
+     * @param {Object} config -  setup the integration test according to the specific scenario and environment the
+     * user wishes to test. If not provided, the defaultConfig is used
+     */
+    initialiseConfig(config) {
+        const defaultConfig = {
+            /**
+             * @param {string[]} contractsToDeploy - Names of the various contracts for which Truffle contracts should be
+             * created to represent those contracts deployed on the relevant testNet
+             */
+            contractsToDeploy: [
+                'ACE',
+                'Dividend',
+                'ERC20Mintable',
+                'FactoryAdjustable201907',
+                'JoinSplit',
+                'JoinSplitFluid',
+                'PrivateRange',
+                'PublicRange',
+                'Swap',
+                'ZkAssetAdjustable',
+            ],
+
+            /**
+             * @constant {string} NETWORK - name of the test network for which the integration test is to be run.
+             * Extracted from the argument passed to the truffle test command - it is the last argument to be passed
+             */
+            NETWORK: process.argv[process.argv.length - 1],
+
+            /**
+             * @constant {BN} numTestTokens - number of ERC20 tokens used during the integration test,
+             * and therefore the required minimum balance of the spending account
+             */
+            numTestTokens: new BN(260),
+
+            /**
+             * @param {Bool} runAdjustSupplyTests - boolean determining whether the adjust supply related integration
+             * tests should be run. Set to true if so, false if not
+             *
+             * Currently necessary as the test suite does not yet support same asset repeated minting/burning
+             */
+            runAdjustSupplyTests: false,
+
+            /**
+             * @param {bool} runUpgradeTest - boolean determining whether the test that performs a note registry upgrade
+             * should be performed. Set to true if so, false if not
+             *
+             */
+            runUpgradeTest: false,
+
+            /**
+             * @param {BN} scalingFactor - factor to convert between AZTEC note value and ERC20 token value
+             */
+            scalingFactor: ERC20_SCALING_FACTOR,
+        };
+
+        if (config) {
+            this.config = config;
+        } else {
+            this.config = defaultConfig;
+        }
```

Comment:
Would replace the above with: 
```suggestion
        this.config = Object.assign({}, defaultConfig, config)
```
Otherwise, someone passing in a config objet needs to pass in every single config key. The above will use the default config, only overwriting the keys passed in through `config`

Reply:
That's a really good idea, have done